index.js

```diff
@@ -1,56 +1,45 @@
 'use strict';
 
-var forEach = require('foreach');
+var toStr = Object.prototype.toString,
+	typedArrays = {
+		Float32Array: true,
+		Float64Array: true,
+		Int8Array: true,
+		Int16Array: true,
+		Int32Array: true,
+		Uint8Array: true,
+		Uint8ClampedArray: true,
+		Uint16Array: true,
+		Uint32Array: true
+	},
+	slice = String.prototype.slice,
+	gOPD = Object.getOwnPropertyDescriptor;
 
-var toStr = Object.prototype.toString;
-var hasToStringTag = typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol';
-
-var typedArrays = {
-	Float32Array: true,
-	Float64Array: true,
-	Int8Array: true,
-	Int16Array: true,
-	Int32Array: true,
-	Uint8Array: true,
-	Uint8ClampedArray: true,
-	Uint16Array: true,
-	Uint32Array: true
-};
-
-var slice = String.prototype.slice;
-var toStrTags = {};
-var gOPD = Object.getOwnPropertyDescriptor;
-if (hasToStringTag && gOPD && Object.getPrototypeOf) {
-	forEach(typedArrays, function (_, typedArray) {
-		var arr = new global[typedArray]();
-		if (!(Symbol.toStringTag in arr)) {
-			throw new EvalError('this engine has support for Symbol.toStringTag, but ' + typedArray + ' does not have the property! Please report this.');
-		}
-		var proto = Object.getPrototypeOf(arr);
-		var descriptor = gOPD(proto, Symbol.toStringTag);
-		if (!descriptor) {
-			var superProto = Object.getPrototypeOf(proto);
-			descriptor = gOPD(superProto, Symbol.toStringTag);
+module.exports = Object.getPrototypeOf && gOPD && typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol' ? function (value) {
+	var typedArray, arr, proto, descriptor;
+	/* eslint max-statements: ['error', 16] */
+	/* eslint no-restricted-syntax: ['warn', 'ForInStatement'] */
+	/* jscs:disable disallowNodeTypes */
+	for (typedArray in typedArrays) {
+		if (Object.prototype.hasOwnProperty.call(typedArrays, typedArray)) {
```

Comment:
yes, it will - it preserves a bound copy in the require cache to prevent things like `delete Object.prototype.hasOwnProperty` from breaking this code.